internal/bminventory/inventory_test.go

```diff
@@ -5665,6 +5665,48 @@ var _ = Describe("cluster", func() {
 			})
 		})
 
+		It("Should handle not found correctly on fetch of UI specific settings for cluster", func() {
+			reply := bm.V2GetClusterUISettings(ctx, installer.V2GetClusterUISettingsParams{
+				ClusterID: clusterID,
+			})
+			Expect(reply).To(BeAssignableToTypeOf(common.NewApiError(http.StatusNotFound, errors.Errorf(""))))
+		})
+
+		It("Should handle not found correctly on update of UI specific settings for cluster", func() {
+			settingsJSON := "{\"foo\":\"bar\"}"
+			reply := bm.V2UpdateClusterUISettings(ctx, installer.V2UpdateClusterUISettingsParams{
+				ClusterID:  clusterID,
+				UISettings: settingsJSON,
+			})
+			Expect(reply).To(BeAssignableToTypeOf(common.NewApiError(http.StatusNotFound, errors.Errorf(""))))
+		})
+
+		It("Should store and retrieve the UI specific settings for a cluster when requested to do so", func() {
+			createCluster(defaultCluster)
+			By("Prior to start of test, settings JSON should be an empty string", func() {
+				reply := bm.V2GetClusterUISettings(ctx, installer.V2GetClusterUISettingsParams{
+					ClusterID: clusterID,
+				})
+				Expect(reply).Should(BeAssignableToTypeOf(installer.NewV2GetClusterUISettingsOK()))
+				Expect(reply.(*installer.V2GetClusterUISettingsOK).Payload).To(Equal(""))
+			})
+			By("Should be able to store UI specific settings for the cluster", func() {
+				settingsJSON := "{\"foo\":\"bar\"}"
+				reply := bm.V2UpdateClusterUISettings(ctx, installer.V2UpdateClusterUISettingsParams{
+					ClusterID:  clusterID,
+					UISettings: settingsJSON,
+				})
+				Expect(reply).Should(BeAssignableToTypeOf(installer.NewV2UpdateClusterUISettingsOK()))
```

Comment:
What are you actually testing here?

This should verify the correct value was set in the database.

Also this should be separate from the `Get` tests. I would expect those to set some state in the DB and then test that the API returns the value correctly.